File: spball/spball.cpp
#include <iostream>
#include <map>
using namespace std;

int main() {
	int t;
	cin >> t;

	map<unsigned long long int, unsigned long long int> facts;
	facts[0] = 1;
	facts[1] = 1;

	while (t--) {
		unsigned long long int n, sum=0;
		cin >> n;

		while (n--) {
			unsigned long long int x;
			cin >> x;

			if (facts.find(x) == facts.end()) {
				for (unsigned long long int i=2; i<=x ; ++i) {
					if (facts.find(i) == facts.end()) {
						facts[i] = facts[i-1] * i;
					}
				}
			}
			sum += facts[x];
		}
		sum %= 1000000007;
		cout << sum << endl;
	}
}

File: spball/spball.py
import sys
sys.set_int_max_str_digits(10**7)
facts = {0:1, 1:1}

for _ in range(int(input())):

    input()
    sum = 0

    for x in input().split():
        n = int(x)

        if n not in facts:
            for i in range(1, n+1):
                if i not in facts:
                    facts[i] = facts[i-1] * i

        sum += facts[n]

    sum = sum % (10**9+7)
    print(sum)
